src/main/java/com/coscale/sdk/client/ApiClient.java

```diff
@@ -319,7 +319,7 @@ private void login() throws IOException {
         // Do the actual request.
         int tries = 0;
         int responseCode = 0;
-        while (tries++ <= AUTH_RETRIES) {
+        do {
```

Comment:
Assuming it has to happen at least once, the do-while consturct is best fitted for that.